js.go

```diff
@@ -1715,13 +1715,13 @@ func (m *Msg) InProgress(opts ...AckOpt) error {
 }
 
 // MsgMetadata is the JetStream metadata associated with received messages.
-type MsgMetaData struct {
-	Consumer   uint64
-	Stream     uint64
-	Delivered  uint64
-	Pending    uint64
-	Timestamp  time.Time
-	StreamName string
+type MsgMetadata struct {
+	StreamSeq    uint64
```

Comment:
ok, trying to look at this in a different way as you both described. I am ok with NumDelivered and NumPending, was wondering if the sequences should there own struct?

```go
type Sequence struct {
  Consumer uint64
  Stream   uint64
}
```

Reply:
Adding the SequencePair sounds good, so would look like this?

```go
type MsgMetadata struct {
	Sequence     SequencePair
	NumDelivered uint64
	NumPending   uint64
	Timestamp    time.Time
	Stream       string
}
```

where `SequencePair` is the same type that is common in other types:

```
type SequencePair struct {
	Consumer uint64 `json:"consumer_seq"`
	Stream   uint64 `json:"stream_seq"`
}
```

https://github.com/wallyqs/nats.go/commit/01f3157fba1687b2890cb6c54599c09321c3d01f

Reply:
Updated to:

```go
type MsgMetadata struct {
	Sequence     SequencePair
	NumDelivered uint64
	NumPending   uint64
	Timestamp    time.Time
	Stream       string
}

meta, _ := m.Metadata()
// Stream sequence
meta.Sequence.Stream
// Consumer sequence
meta.Sequence.Consumer
```